File: app/src/main/java/com/example/a_notes/ui/NotesAdapter.java
package com.example.a_notes.ui;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.a_notes.domain.NoteEntity;

import java.util.ArrayList;
import java.util.List;

public class NotesAdapter extends RecyclerView.Adapter<NoteViewHolder> {

    private List<NoteEntity> data = new ArrayList<>();
    private OnItemClickListener clickListener;

    public NotesAdapter (OnItemClickListener clickListener){
        this.clickListener = clickListener;
    }

    public void setData(List<NoteEntity> data) {
        this.data = data;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public NoteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new NoteViewHolder(parent, clickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull NoteViewHolder holder, int position) {
        holder.bind(getItem(position));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    private NoteEntity getItem(int position) {
        return data.get(position);
    }
}

File: app/src/main/java/com/example/a_notes/ui/NotesAdapter.kt
package com.example.a_notes.ui

import androidx.recyclerview.widget.RecyclerView
import com.example.a_notes.ui.NoteViewHolder
import com.example.a_notes.domain.NoteEntity
import android.view.ViewGroup
import java.util.ArrayList

class NotesAdapter(private val clickListener: OnItemClickListener) :
    RecyclerView.Adapter<NoteViewHolder>() {

    private var data: List<NoteEntity> = ArrayList()
    fun setData(data: List<NoteEntity>) {
        this.data = data
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteViewHolder {
        return NoteViewHolder(parent, clickListener)
    }

    override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    override fun getItemCount(): Int {
        return data.size
    }

    private fun getItem(position: Int): NoteEntity {
        return data[position]
    }
}